var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -311,20 +302,15 @@ def setup_build_environment(self, env):
         ##############
         # ParallelIO #
         ##############
-        if "+parallelio" in spec:
+        if "+external-parallelio" in spec:
             env.set("ESMF_PIO", "external")
             env.set("ESMF_PIO_LIBPATH", spec["parallelio"].prefix.lib)
             env.set("ESMF_PIO_INCLUDE", spec["parallelio"].prefix.include)
-        elif "+pio" in spec and "+mpi" in spec:
-            # ESMF provides the ability to read and write data in both binary
-            # and NetCDF formats through ParallelIO (PIO), a third-party IO
-            # software library that is integrated in the ESMF library.
-            # PIO-dependent features will be enabled and will use the
-            # PIO library that is included and built with ESMF.
-            env.set("ESMF_PIO", "internal")
-        else:
-            # Disables PIO-dependent code.
-            env.set("ESMF_PIO", "OFF")
+
+        # Allow override of ESMF_PIO:
```

Comment:
How this works? Do we need to set `esmf_pio` in the spec if we want to use internal PIO?

Reply:
Not for anything newer than v8.3.1, it is automatic when you select `~external-parallelio`.

Reply:
@uturuncoglu @theurich I feel pretty strongly that newer versions of ESMF should **never** use the internal build of PIO in spack.   It is too easy when using ESMF with CESM (for example) to end up with two different PIO versions which leads to a great deal of confusion.   Since spack easily handles the external pio build why would you ever want it to be internal?

Reply:
I agree with @jedwards4b

Reply:
We want to use Spack also for regression testing by the ESMF core team. In fact @uturuncoglu has started setting up GitHub actions to do this. For this we want to control turning off the default `+external-parallelio`.

Reply:
Why not build the parallelio external for the github action as well?  That would allow you to cache the pio build and get a faster turnaround on the already very slow esmf build process.

Reply:
Yes, of course, we test that. But @uturuncoglu would have more details on that. But we _also_ want to test ESMF_PIO=internal. Therefore we need to be able to set `~external-parallelio`. We just want the _option_, we are not discussing the default.